staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -76,38 +77,85 @@ const (
 	zonesKey = "zones"
 )
 
-// replaceTopology overwrites an existing topology key by a new one.
+// replaceTopology overwrites an existing key in NodeAffinity by a new one.
+// If there are any newKey already exist in a term, we will combine the
+// replaced key Values with the existing ones
```

Comment:
Ohh, currently in my implementation I will combine them together... So basically if we have
`gke.topology.io/zone=zone1` and `topology.kubernetes.io/zone=zone2`
I will combine it to `topology.kubernetes.io/zone=zone1 and zone2`

Reply:
hmmmm I think that is changing the semantics of the NodeAffinity.

If they're combined in the same expression, then it is ORed together, which means the volume must be in "zone1 OR zone2".

If they're in different expressions, we AND them together, which means the volume must be in "zone1 AND zone2". Which at least with the volumes we're dealing with, would not make sense.

Reply:
Thinking about this more, maybe we should not try to reduce terms here. It would be very strange (and probably non functional) if a user created a PV like this. Instead of trying to fix their problem, we can just leave it as is and let them fix it. And in the case of where it's the same values, having 2 ANDs of the same thing should be fine (we can also try it out to make sure the scheduler handles it properly)

Reply:
> If they're combined in the same expression, then it is ORed together, which means the volume must be in "zone1 OR zone2". If they're in different expressions, we AND them together, which means the volume must be in "zone1 AND zone2". 

Yeah, this makes sense to me. I did not think of that and that would definitely be a weird case. Hmm, not remove duplication is definitely not so UX friendly. But think about this more, it only get called in the csi-provisioner and right now we do not add the Kubernetes topology there so maybe we are over thinking. I will just remove the whole logic then. Sounds like we just need to replace it.

Reply:
Done